### PR TITLE
Handle authentication type in Facebook

### PR DESCRIPTION
**Pull request by nuvipannu**
Allow using authType in Facebook login (https://developers.facebook.com/docs/facebook-login/reauthentication) :balloon:

### CODE REVIEW

**Comment by timbotnik** [comment]
Seems useful, thanks!

**Comment by IonicaBizau** [comment]
@nuvipannu  :+1: :sparkles: 

@timbotnik Take a look at #5693 as well. :grin: While coding together with @nuvipannu, we discovered these limitations and thought it's a good thing for universe to fix them. For some reason the tests are failing, but doesn't seem to be related to our changes. :dizzy:

**Comment by robertpitt** [comment]
You will probably need to update the documentation to show the new option before this is accepted.

**Comment by IonicaBizau** [comment]
@robertpitt Where should I update the docs for these OAuth APIs? :thought_balloon:

**Comment by IonicaBizau** [comment]
@robertpitt The relevant docs I found [are these](http://docs.meteor.com/#/basic/accounts)––but they don't contain the specific facebook/twitter information.

**Comment by IonicaBizau** [comment]
@benjamn Can you take a look at this pull request and #5693? :bell:

**Comment by JonathanHindi** [comment]
@benjamn What's pending for this to be merged?

**Comment by IonicaBizau** [comment]
@benjamn @JonathanHindi I'm pretty sad to see how much time it takes for a simple fix like this to get merged (actually, this is not even merged yet). :disappointed:

File: packages/facebook/facebook_client.js [hunk 1]
```diff
@@ -35,7 +35,12 @@ Facebook.requestCredential = function (options, credentialRequestCompleteCallbac
         '&redirect_uri=' + OAuth._redirectUri('facebook', config) +
         '&display=' + display + '&scope=' + scope +
         '&state=' + OAuth._stateParam(loginStyle, credentialToken, options && options.redirectUrl);
-
+  
+  // Handle authentication type (e.g. for force login you need authType: "reauthenticate")
+  if (options.authType) {
+    loginUrl = loginUrl + "&authType=" + options.authType;
```
Comment (benjamn) [inline packages/facebook/facebook_client.js]: I think this should maybe be `loginUrl += "&authType=" + encodeURIComponent(options.authType)` to make sure the `authType` is properly URL-escaped?

Comment (IonicaBizau) [inline packages/facebook/facebook_client.js]: @benjamn I don't think that's really needed, but just in case, I just pushed it.

**Comment by IonicaBizau** [comment]
@JonathanHindi @benjamn Any ideas why the tests are failing? 💥 

Same happens in #6987.
I have the feeling it doesn't happen because of my (or @nuvipannu's) changes. 🚀

**Comment by zol** [comment]
Hey @nuvipannu and @JonathanHindi . Sorry it's taken us so long to look at this.We're now working through our PR backlog and merging whatever we can, this looks like a good candidate. Could you please rebase against the latest `devel` branch as we've recently fixed our CI infrastructure so tests should now be passing (as you said, it's unlikely this PR would have broken any test).

**Comment by IonicaBizau** [comment]
@zol Thanks for feedback. I have just rebased and pushed. Waiting for the tests. :hourglass_flowing_sand:

**Comment by IonicaBizau** [comment]
@zol @nuvipannu Tests are passing :tada: :rocket:

**Comment by zol** [comment]
Merged, thank you.

**Comment by IonicaBizau** [comment]
@nuvipannu Finally! Yay! :tada: :grin: :confetti_ball:

**Comment by sahanDissanayake** [comment]
is this documented on the Guide ? thanks

**Pull request merged**